pkg/scalers/azure_monitor_scaler.go

```diff
@@ -0,0 +1,189 @@
+		return nil, fmt.Errorf("error parsing azure monitor metadata: %s", err)
+	}
+
+	return &azureMonitorScaler{
+		metadata: meta,
+	}, nil
+}
+
+func parseAzureMonitorMetadata(metadata, resolvedEnv, authParams map[string]string) (*azureMonitorMetadata, error) {
+	meta := azureMonitorMetadata{}
+
+	if val, ok := metadata[targetValueName]; ok && val != "" {
+		targetValue, err := strconv.Atoi(val)
+		if err != nil {
+			azureMonitorLog.Error(err, "Error parsing azure monitor metadata", "targetValue", targetValueName)
+			return nil, fmt.Errorf("Error parsing azure monitor metadata %s: %s", targetValueName, err.Error())
+		}
+		meta.targetValue = targetValue
+	} else {
+		return nil, fmt.Errorf("no targetValue given")
+	}
+
+	if val, ok := metadata["resourceURI"]; ok && val != "" {
+		resourceURI := strings.Split(val, "/")
+		if len(resourceURI) != 3 {
+			return nil, fmt.Errorf("resourceURI not in the correct format. Should be namespace/resource_type/resource_name")
+		}
+		meta.resourceURI = val
+	} else {
+		return nil, fmt.Errorf("no resourceURI given")
+	}
+
+	if val, ok := metadata["resourceGroupName"]; ok && val != "" {
+		meta.resourceGroupName = val
+	} else {
+		return nil, fmt.Errorf("no resourceGroupName given")
+	}
+
+	if val, ok := metadata[azureMonitorMetricName]; ok && val != "" {
+		meta.name = val
+	} else {
+		return nil, fmt.Errorf("no metricName given")
+	}
+
+	if val, ok := metadata["metricAggregationType"]; ok && val != "" {
+		meta.aggregationType = val
+	} else {
+		return nil, fmt.Errorf("no metricAggregationType given")
+	}
+
+	if val, ok := metadata["metricFilter"]; ok && val != "" {
+		meta.filter = val
+	}
+
+	if val, ok := metadata["metricAggregationInterval"]; ok && val != "" {
+		aggregationInterval := strings.Split(val, ":")
+		if len(aggregationInterval) != 3 {
+			return nil, fmt.Errorf("metricAggregationInterval not in the correct format. Should be hh:mm:ss")
+		}
+	}
+
+	// Required authentication parameters below
+
+	if val, ok := authParams["subscriptionId"]; ok && val != "" {
+		meta.subscriptionID = val
+	} else {
+		if val, ok := metadata["subscriptionId"]; ok && val != "" {
+			meta.subscriptionID = val
+		} else {
+			return nil, fmt.Errorf("no subscriptionId given")
+		}
+	}
+
+	if val, ok := authParams["tenantId"]; ok && val != "" {
+		meta.tenantID = val
+	} else {
+		if val, ok := metadata["tenantId"]; ok && val != "" {
+			meta.tenantID = val
+		} else {
+			return nil, fmt.Errorf("no tenantId given")
+		}
+	}
+
+	if val, ok := authParams["activeDirectoryClientId"]; ok && val != "" {
+		meta.clientID = val
+	} else {
+		if val, ok := metadata["activeDirectoryClientId"]; ok && val != "" {
+			meta.clientID = val
+		} else {
+			return nil, fmt.Errorf("no activeDirectoryClientId given")
+		}
+	}
+
+	if val, ok := authParams["activeDirectoryClientPassword"]; ok && val != "" {
+		meta.clientPassword = val
+	} else {
+		if val, ok := metadata["activeDirectoryClientPassword"]; ok && val != "" {
```

Comment:
we shouldn't read the password from metadata.Either from `authParams` or `resolvedEnv` since they are both backed by a `Secret` object. Supporting only one of the 2 is fine, but would rather not have the options of storing a password in metadata in clear text.

Reply:
Agreed, let's use authParams maybe?

Reply:
Supporting just `authParams` sounds like a good idea. @tomkerkhove @ahmelsayed, I'm hearing that for the service principal I should not support plain text. What do we think about `tenantId` and `subscriptionId`? Should I support putting them in plain text AND in `authParams`? Only in plain text? Only in `authParams`?

Reply:
I'd say supporting both would be good imo

Reply:
@tomkerkhove, thoughts on whether we can just put `tenantId` and `subscriptionId` as plain text in the ScaledObject? imo, there's no reason they need to be secrets.

Reply:
Fine by me